### PR TITLE
Onboarding Guide a.k.a. Roadmap

### LINKED ISSUE #219: Provide instructions on how to connect to sensitive data
**Is your feature request related to a problem? Please describe.**
Currently, we propose various templates for provisioning FL setups. However, we do not explicitly state all the options for connecting sensitive data.

**Describe the solution you'd like**
A single document summarizing the various ways to connect sensitive data (manually give permissions to the silos computes, use silo template leveraging an existing storage account...)

### PR DESCRIPTION
**Pull request by thomasp-ms**
## Purpose

* This PR adds a high-level onboarding guide (the roadmap), along with explicit instructions for connecting sensitive data. It closes the following:
* #219 
* #222 
* #245 

## What is the expected review turnaround time?

Urgency:
- [x] Medium (needs review within a few days - most common)

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[x] Documentation content changes
[ ] Other... Please describe:
```

## What to Check
Verify that the following are valid
* Make sure the instructions are understandable, well written, and accurate.

### CODE REVIEW

File: docs/roadmap.md
Comment (jfomhover) [inline docs/roadmap.md:1]: Can we replace "roadmap" by "onboarding guide" ? the term Roadmap might give the impression we're talking about the roadmap of the AzureML platform towards FL

Comment (jfomhover) [reply docs/roadmap.md:1]: or "FL ramp-up plan"

File: docs/roadmap.md
Comment (jfomhover) [inline docs/roadmap.md:5]: can we add a section on who's the audience for this tutorial? and what value they'll find in the page?

(like learning goals for different personas: what a ML/DS engineer will find interesting, what a team lead will find, etc?)

File: docs/roadmap.md
Comment (jfomhover) [inline docs/roadmap.md:8]: For each of the phases, let's formalize what is the starting point, who will be involved, and what outcome to expect.

Example here:

This will likely apply to any organization that hasn't done any Federated Learning training before, or that has practiced some FL using available open source frameworks in a simulated environment.

During this phase, we recommend you to get used to both the scientific and the infrastructure aspects of FL.

Do the following:

`<then insert a list of tutorials to try>`

File: docs/roadmap.md
Comment (jfomhover) [inline docs/roadmap.md:14]: I think there's a component missing here: security review.

Maybe add a guideline that it's time to start discussing with your own security team about integration constraints that might prevent your progress past the experimentation phase.

File: docs/roadmap.md
Comment (jfomhover) [inline docs/roadmap.md:24]: first FL model => + "relevant to your business goals"

the feasibility of the approach => + "and the value for your organization"

File: docs/roadmap.md
Comment (jfomhover) [inline docs/roadmap.md:16]: I'd suggest create 2 tracks within this phase (under 2 sections), one for ramping up on the ML/DS side using synthetic data, the other on ramping up the infra requirements, with custom setup, etc.

File: docs/roadmap.md
Comment (jfomhover) [inline docs/roadmap.md:26]: this phase could end with a parity check, demonstrating the value of the FL model on prod data versus centralized / legacy model.

then you would use the Fly phase to organize FL experiments, iterate on the design and increase (we could add some of the AzureML feature to help with experiment management)

**Review by jfomhover — CHANGES_REQUESTED** [review]
Thank you for putting this draft together! I added some comments, happy to talk about it in detail.

File: docs/roadmap.md
Comment (thomasp-ms) [reply docs/roadmap.md:26]: Added the parity check.

About the Fly phase (MLOps). We haven't considered this part much yet. Should we have this PR wait on the Fly phase, or should we just release Crawl-Walk-Run and get to Fly later? (I have a slight preference towards the second option because it would let us release the current content ASAP as opposed to having to rush going through the Fly phase). Let me know what you think!

File: docs/roadmap.md
Comment (thomasp-ms) [reply docs/roadmap.md:24]: done

File: docs/roadmap.md
Comment (thomasp-ms) [reply docs/roadmap.md:5]: done. please take another look

File: docs/roadmap.md
Comment (thomasp-ms) [reply docs/roadmap.md:8]: I followed this structure for the 3 phases. Please have a look

File: docs/roadmap.md
Comment (thomasp-ms) [reply docs/roadmap.md:16]: done

File: docs/ramp-up-plan.md
Comment (thomasp-ms) [inline docs/ramp-up-plan.md:48]: note to self: need to clarify the scenario brought up by Chengyu (AKS compute in an external Azure tenant, data in storage account in same tenant)

**Comment by jfomhover** [comment]
Just merged from release-05 to resolve conflicts with #250

**Review by jfomhover — CHANGES_REQUESTED** [review]
a couple of typos and suggestions

File: docs/concepts/plan-your-fl-project.md
Comment (thomasp-ms) [reply docs/concepts/plan-your-fl-project.md:73]: good eye!

**Review by jfomhover — APPROVED** [review]

**Pull request merged**
